### PR TITLE
Update the stub code of BOSS

### PR DESCRIPTION
**Pull request by JamePeng**
I redo the #2019 PR and append a lot of stub code and comment which was investigated through reverse engineering the boss module.  But there are many parameter meaning was unknown (>m<), sorry!

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-08-28T17:22:48Z: https://github.com/citra-emu/citra/compare/950f6c0c200fd6a91131f088a68f9605ebd925e8...JamePeng:7344f6e43ca2a012477a70af8cb98522e224468e

File: src/core/hle/service/boss/boss.cpp [hunk 1]
```diff
@@ -2,19 +2,939 @@
...
+    u32 translation = cmd_buff[3];
+    u32 buff_addr = cmd_buff[4];
+    u32 buff_size = (translation >> 4);
+
+    cmd_buff[0] = IPC::MakeHeader(0x17, 0x1, 0x2);
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = (buff_size << 4 | 0xA);
+    cmd_buff[3] = buff_addr;
+
+    LOG_WARNING(Service_BOSS, "(STUBBED)unk_param1=0x%08X, unk_param2=0x%08X, translation=0x%08X, buff_addr=0x%08X, buff_size=0x%08X",
```
Comment (bunnei) [inline src/core/hle/service/boss/boss.cpp]: nite: Missing space after (STUBBED)

File: src/core/hle/service/boss/boss.cpp [hunk 2]
```diff
@@ -2,19 +2,939 @@
...
+    u32 translation = cmd_buff[4];
+    u32 buff_addr = cmd_buff[5];
+    u32 buff_size = (translation >> 4);
+
+    cmd_buff[0] = IPC::MakeHeader(0xB, 0x1, 0x2);
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = (buff_size << 4 | 0xA);
+    cmd_buff[3] = buff_addr;
+
+    LOG_WARNING(Service_BOSS, "(STUBBED)unk_param1=0x%08X, unk_param2=0x%08X, unk_param3=0x%08X, translation=0x%08X, buff_addr=0x%08X, buff_size=0x%08X",
```
Comment (bunnei) [inline src/core/hle/service/boss/boss.cpp]: nit: Missing space after (STUBBED)... Will stop pointing out, please fix them all!

File: src/core/hle/service/boss/boss.cpp [hunk 3]
```diff
@@ -2,19 +2,939 @@
...
+    // TODO(JamePeng): Figure out the meaning of these parameters
+    u32 unk_param1 = cmd_buff[1];
+    u32 unk_param2 = cmd_buff[2];
+    u32 unk_param3 = cmd_buff[3];
+    u32 unk_param4 = cmd_buff[4];
+    u32 translation = cmd_buff[5];
+    u32 buff_addr = cmd_buff[6];
+    u32 buff_size = (translation >> 4);
+
+
```
Comment (bunnei) [inline src/core/hle/service/boss/boss.cpp]: nit: Extra space

File: src/core/hle/service/boss/boss.cpp [hunk 4]
```diff
@@ -2,19 +2,939 @@
...
+void GetTaskInterval(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    u32 unk_param1 = cmd_buff[1];  // TODO(JamePeng): Figure out the meaning of these parameters
+    u32 translation = cmd_buff[2];
+    u32 buff_addr = cmd_buff[3];
+    u32 buff_size = (translation >> 4);
+
+
```
Comment (bunnei) [inline src/core/hle/service/boss/boss.cpp]: nit: Extra line, will stop pointing out -- please fix the other occurances

**Comment by bunnei** [comment]
This might be the biggest noop change in Citra history... Nonetheless, code looks fine assuming my nits are addressed. Thanks!

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-06T08:20:18Z: https://github.com/citra-emu/citra/compare/c3307b41db4e222ceca3860c75ccbe6300484729...JamePeng:dd0b31f3f43788d34fa76dcd9eb3d8e92a35c539

**Comment by JamePeng** [comment]
Addressed

File: src/core/hle/service/boss/boss.h [hunk 5]
```diff
@@ -4,9 +4,795 @@
...
+ *      2 : MappedBufferDesc(permission = R)
+ *      3 : u32 buff_addr
+ *  Outputs:
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : buff_size << 4 | 0xA
+ *      3 : u32 buff_addr
+ */
+void StartBgImmediate(Service::Interface* self);
+
+
```
Comment (Dragios) [inline src/core/hle/service/boss/boss.h]: Unnecessary newline

File: src/core/hle/service/boss/boss.h [hunk 6]
```diff
@@ -4,9 +4,795 @@
...
+/**
+ * BOSS::RegisterPrivateRootCa service function
+ *  Inputs:
+ *      0 : Header Code[0x00050042]
+ *      1 : u32 unknown value
+ *      2 : MappedBufferDesc(permission = R)
+ *      3 : u32 buff_addr
+ *  Outputs:
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : buff_size << 4 | 0xA
```
Comment (wwylele) [inline src/core/hle/service/boss/boss.h]: (Un)MappedBufferDesc(permission = R) matching the inputs. Same as below

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-06T15:16:39Z: https://github.com/citra-emu/citra/compare/c3307b41db4e222ceca3860c75ccbe6300484729...JamePeng:db80061d35430b065c7715a023e982cf74bd79e5

File: src/core/hle/service/boss/boss.h [hunk 7]
```diff
@@ -4,9 +4,794 @@
 
 #pragma once
 
+#include "core/hle/service/service.h"
```
Comment (mailwl) [inline src/core/hle/service/boss/boss.h]: why include here?
do as here https://github.com/citra-emu/citra/blob/master/src/core/hle/service/cecd/cecd.h#L9

**Pull request merged**